main.go

```diff
@@ -186,7 +186,7 @@ func main() {
 		}
 
 		defer func() {
-			closeCtx, cancelClose := context.WithTimeout(ctx, rootConf.RequestTimeout)
+			closeCtx, cancelClose := context.WithTimeout(context.Background(), rootConf.RequestTimeout)
 			defer cancelClose()
```

Comment:
Why context Background here?

Reply:
Because we should not use app context.
In this case on POD deleting the client will not close his connection.

Reply:
I can see not using the *same* context (as noted, it will be expired)... but we should have *some* timeout on the Close... perhaps note the amount of time left on the 'request' ctx and use that as duration for the new context for Close?

Reply:
I read this too quickly, this is exactly write as currently written.